integration/concurrency_sqlite/concurrency_sqlite_test.go

```diff
@@ -0,0 +1,171 @@
+package concurrency_sqlite_test
+
+import (
+	"fmt"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	_ "github.com/mattn/go-sqlite3"
+
+	boshlog "github.com/cloudfoundry/bosh-utils/logger"
+	common "github.com/maximilien/bosh-softlayer-cpi/common"
+	bslcvmpool "github.com/maximilien/bosh-softlayer-cpi/softlayer/vm/pool"
+	testhelpers "github.com/maximilien/softlayer-go/test_helpers"
+)
+
+var (
+	SQLITE_DB_FOLDER    = "/tmp/concurrency_sqlite_test"
+	SQLITE_DB_FILE      = "vm_pool.sqlite"
+	SQLITE_DB_FILE_PATH = filepath.Join(SQLITE_DB_FOLDER, SQLITE_DB_FILE)
+	stemcellUuid        = "fake_stemcell_uuid"
+	domain              = "softalyer.com"
+	c                   = make(chan string, 0)
+	logger              = boshlog.NewLogger(boshlog.LevelInfo)
+)
+
+func populateDB() {
+	db, err := bslcvmpool.OpenDB(SQLITE_DB_FILE_PATH)
+	Expect(err).ToNot(HaveOccurred())
+
+	for i := 0; i < 1000; i++ {
+		vmID := i
+		hostname := fmt.Sprintf("concurrency_test_%d", vmID)
+		agentID := strconv.Itoa(i)
+		vmInfoDB := bslcvmpool.NewVMInfoDB(vmID, hostname+"."+domain, "t", stemcellUuid, agentID, logger, db)
+		defer vmInfoDB.CloseDB()
+
+		err = vmInfoDB.InsertVMInfo(bslcvmpool.DB_RETRY_TIMEOUT, bslcvmpool.DB_RETRY_INTERVAL)
+		Expect(err).ToNot(HaveOccurred())
+	}
+}
+
+func insertVMInfo(init int) {
+	db, err := bslcvmpool.OpenDB(SQLITE_DB_FILE_PATH)
+	Expect(err).ToNot(HaveOccurred())
+
+	for i := init; i < init+1000; i++ {
+		vmID := i
+		hostname := fmt.Sprintf("concurrency_test_%d", vmID)
+		agentID := strconv.Itoa(i)
+		vmInfoDB := bslcvmpool.NewVMInfoDB(vmID, hostname+"."+domain, "t", stemcellUuid, agentID, logger, db)
+		//defer vmInfoDB.CloseDB()
+
+		err = vmInfoDB.InsertVMInfo(bslcvmpool.DB_RETRY_TIMEOUT, bslcvmpool.DB_RETRY_INTERVAL)
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	c <- "Done!"
+}
+
+func updateVMInfoByID() {
+	db, err := bslcvmpool.OpenDB(SQLITE_DB_FILE_PATH)
+	Expect(err).ToNot(HaveOccurred())
+
+	for i := 0; i < 100; i++ {
+		vmID := rand.Intn(100)
+		hostname := fmt.Sprintf("concurrency_test_%d", vmID)
+		agentID := strconv.Itoa(i)
+		vmInfoDB := bslcvmpool.NewVMInfoDB(vmID, hostname+"."+domain, "f", stemcellUuid, agentID, logger, db)
+		defer vmInfoDB.CloseDB()
+
```

Comment:
add random sleep between 1 and 100 milliseconds so that it behaves more like a reallife use case